lib/node_modules/@stdlib/math/base/special/sqrt1pm1/test/test.native.js

```diff
@@ -0,0 +1,126 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+
+
+// FIXTURES //
+
+var large = require( './fixtures/cpp/large.json' );
+var small = require( './fixtures/cpp/small.json' );
+
+
+// VARIABLES //
+
+var sqrt1pm1 = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( sqrt1pm1 instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof sqrt1pm1, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', opts, function test( t ) {
+	var v = sqrt1pm1( NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `+infinity` if provided `+infinity`', opts, function test( t ) {
+	var v = sqrt1pm1( PINF );
+	t.equal( v, PINF, 'returns +infinity' );
+	t.end();
+});
+
+tape( 'the function returns `0` if provided `0`', opts, function test( t ) {
+	var v = sqrt1pm1( 0.0 );
+	t.equal( v, 0.0, 'returns zero' );
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided `-1`', opts, function test( t ) {
+	var v = sqrt1pm1( -1.0 );
+	t.equal( v, -1.0, 'returns negative one' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided a `x < -1`', opts, function test( t ) {
+	var v = sqrt1pm1( -1.5 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	sqrt1pm1( -2.0 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	sqrt1pm1( -3.0 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function handles large `x` values (|x| > 0.75)', opts, function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = large.x;
+	expected = large.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = sqrt1pm1( x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 1.0 * EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. E: ' + expected[ i ] + '. Δ: ' + delta + '. tol: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function handles small `x` values (|x| <= 0.75)', opts, function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = small.x;
+	expected = small.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = sqrt1pm1( x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = 390.0 * EPS * abs( expected[ i ] );
```

Comment:
This is a relatively high tolerance. Increased relative error due to either `expm1` or `log1p`. Matches JS implementation tests, so fine to use here.